e2e/test/scenarios/permissions/create-queries.cy.spec.js

```diff
@@ -279,6 +287,138 @@ describe("scenarios > admin > permissions > create queries > query builder and n
 
     assertPermissionTable(finalPermissions);
   });
+
+  it("should allow setting 'query builder and native' in table view and it should be enforced correctly", () => {
+    cy.log("should allow setting permissions value at table level");
+
+    cy.visit(`/admin/permissions/data/group/${DATA_GROUP}`);
+
+    const CREATE_QUERIES_PERM_IDX = 0;
+
+    modifyPermission(
+      "Sample Database",
+      CREATE_QUERIES_PERM_IDX,
+      "Query builder only",
+    );
+
+    cy.get("main").findByText("Sample Database").click();
+
+    modifyPermission(
+      "Orders",
+      CREATE_QUERIES_PERM_IDX,
+      "Query builder and native",
+    );
+
+    assertSameBeforeAndAfterSave(() => {
+      assertPermissionForItem(
+        "Accounts",
+        CREATE_QUERIES_PERM_IDX,
+        "Query builder only",
+      );
+      assertPermissionForItem(
+        "Orders",
+        CREATE_QUERIES_PERM_IDX,
+        "Query builder and native",
+      );
+    });
+
+    cy.log("should enforce the new settings");
+    cy.signOut();
+    cy.signInAsNormalUser();
+
+    cy.log("should work for table with native query access");
+    startNewNativeQuestion();
+    focusNativeEditor().type("SELECT * FROM Orders");
+    runNativeQuery();
+    cy.findAllByTestId("cell-data").should("contain", "1");
```

Comment:
nit, but it seems real lucky that there's only one cell with a 1? can we maybe think of something a little more unique to test that the query succeeds?